cfme/networks/topology.py

```diff
@@ -0,0 +1,200 @@
+import re
+import time
+
+from navmazing import NavigateToAttribute
+
+from cfme.networks.topology_view import TopologyView
+from cfme.utils.appliance import Navigatable
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from wait_for import wait_for
+
+
+class Topology(Navigatable):
+    LEGENDS = '//kubernetes-topology-icon'
+    ELEMENTS = '//kubernetes-topology-graph//*[name()="g"]'
+    LINES = '//kubernetes-topology-graph//*[name()="lines_obj"]'
+
+    def __init__(self, appliance):
+        Navigatable.__init__(self, appliance=appliance)
+        self.legends_obj = []
+        self.elements_obj = []
+        self.lines_obj = []
+        self.element_ref = None
+        self.display_names = None
+        self.view = None
+
+        self.reload_view()
+        self.refresh()
+        self.reload_elements_and_lines()
+        self.reload_legends()
+
+    def __repr__(self):
+        return "<Topology {}>".format(", ".join(self._legends_obj))
+
+    def reload_view(self):
+        self.view = navigate_to(self, 'All')
+
+    def refresh(self):
+        self.view.toolbar.refresh.click()
+        time.sleep(5)
+        self.reload_elements_and_lines()
+
+    def reload_elements_and_lines(self):
```

Comment:
**REQURIED**
I guess there should be TopologyElementCollection and it should have relationship with `TopologyCollection`

Reply:
@mfalesni Should there be the collections for topology elements and lines?